src/main/java/seedu/address/model/tag/PeriodTag.java

```diff
@@ -47,6 +47,8 @@ public class PeriodTag extends Tag {
      * @return {@code true} if both the period and year are valid; {@code false} otherwise.
      */
     private static boolean isValidPeriod(String period, int year) {
+        assert period != null : "period should not be null";
```

Comment:
Great job on using assertions!

Reply:
Could consider using the assert not null method from JUnit